limit sphinx to `< 7.2.0` in requirements-relaxed

gotmax23 opened the pull request:
limit sphinx to `< 7.2.0` in requirements-relaxed

7.2.0 breaks sphinx-notfound-page. This commit sets a general constraint
that applies to the testing requirements (requirements-relaxed).

Relates: https://github.com/ansible/ansible-documentation/issues/327

gotmax23 committed:
limit sphinx to `< 7.2.0` in requirements-relaxed

7.2.0 breaks sphinx-notfound-page. This commit sets a general constraint
that applies to the testing requirements (requirements-relaxed).

Relates: https://github.com/ansible/ansible-documentation/issues/327

gotmax23 commented:
Rebased to fix commit message

samccann reviewed (changes requested):
just putting a hold so we can decide which PR to go forward with, this or https://github.com/ansible/ansible-documentation/pull/320 

and then test/verify the results of a jenkins build

gotmax23 commented:
> just putting a hold so we can decide which PR to go forward with, this or #320

These PRs address separate issues. This PR sets a general constraint that applies to the testing requirements (requirements-relaxed). Adding an exact pin for the sphinx version is a separate issue and handled by @oranod's PR. The two PRs are additive, not mutually exclusive.

> and then test/verify the results of a jenkins build

Yeah, even though the relaxed requirements file is not used for production docs builds, it'd still be a good idea to make sure the build works on testing.

samccann commented:
Ok thanks. Is there an order in which these PRs should merge or does it not matter?

gotmax23 commented:
> Ok thanks. Is there an order in which these PRs should merge or does it not matter?

It doesn't matter, but I would merge this one first. I changed it a bit so all it does is set a general constraint and not update lock files at all. Therefore, it shouldn't affect docs builds at all, as they have a specific sphinx version pinned.

I also added some PR body text. Thanks for the suggestion.

samccann reviewed (approved):
LGTM. Let's wait for two approvals before merging

oraNod commented:
Thanks @gotmax23 and all
